File: stacks/150_evaluate_reverse_polish_notation.py
from typing import List, Set

class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operators: Set[str] = set(["+", "-", "*", "/"])
        stack: List[int] = list()
        for token in tokens:
            if token in operators:
                right: int = stack.pop()
                left: int = stack.pop()
                match token:
                    case "+":
                        stack.append(left + right)
                    case "-":
                        stack.append(left - right)
                    case "*":
                        stack.append(left * right)
                    case "/":
                        stack.append(int(left / right))
            else:
                stack.append(int(token))
        return stack[0]

File: stacks/150_evaluate_reverse_polish_notation.cpp
#include <stack>
#include <string>
#include <unordered_set>

class Solution {
public:
    int evalRPN(std::vector<std::string>& tokens) {
        std::stack<int> values;
        std::unordered_set<std::string> operators {std::string {"+"}, std::string {"-"}, std::string {"*"}, std::string {"/"}};
        for (std::string& token : tokens) {
            if (operators.find(token) != operators.end()) {
                int right = values.top();
                values.pop();
                int left = values.top();
                values.pop();
                if (token == "+") {
                    values.push(left + right);
                }
                else if (token == "-") {
                    values.push(left - right);
                }
                else if (token == "*") {
                    values.push(left * right);
                }
                else {
                    values.push(static_cast<int>(left / right));
                }
            }
            else {
                values.push(std::stoi(token));
            }
        }
        return values.top();
    }
};